### ISSUE TITLE
shields: Can't use multiple shields anymore 

### ISSUE THREAD

**Issue by erwango** [zephyrproject-rtos/zephyr#25656]
**Describe the bug**
As described in https://docs.zephyrproject.org/latest/guides/porting/shields.html?highlight=shields#shield-activation, it used to be possible to use more than one shield in build command:
`west build -b None your_app -- -DSHIELD="x_nucleo_idb05a1 x_nucleo_iks01a1"`

This doesn't work anymore:
```
$ west build -b nucleo_f401re -- -DSHIELD="x_nucleo_idb05a1 x_nucleo_iks01a1"
-- west build: generating a build system
Including boilerplate (Zephyr base (cached)): /local/mcu/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /local/mcu/zephyrproject/zephyr/samples/hello_world
-- Zephyr version: 2.3.0-rc1 (/local/mcu/zephyrproject/zephyr)
-- Board: nucleo_f401re
No shield named 'x_nucleo_idb05a1 x_nucleo_iks01a1' found
```

**Expected behavior**
Use of multiple shields should be possible

**Impact**
Block multiple shields application (eth + sensors, ...)

**Comment by erwango** [comment]
@tejlmand, can you have a look ?

**Comment by tejlmand** [comment]
Looking into this.
@erwango until the reason for this to not work when using space separated lists, you can use a proper CMake ;-separated list: 
```
west build -b nucleo_f401re -- -DSHIELD="x_nucleo_idb05a1;x_nucleo_iks01a1"
```
which will work.

**Comment by erwango** [comment]
@tejlmand as this command is working, I'm totally fine to update the doc and get away with this.
I'm not particularly attached to the space separated list ....

**Comment by tejlmand** [comment]
> @tejlmand as this command is working, I'm totally fine to update the doc and get away with this.
> I'm not particularly attached to the space separated list ....

As a long time CMake user, i'm familiar with ;-separated lists, but in Zephyr, there seem to be a tradition to also allow space-separated lists as arguments.

And because of that, and to stay consistent with `BOARD_ROOT`, `DTS_ROOT`, etc, I think we should still support space-separated lists.

I only made the comment as a work around (or the CMake way :wink: ), as I wasn't aware how easy it would be to fix.
But the issue is fixed here: #25716
